File: client/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpNearbyEntitiesData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpNearbyEntitiesData(
        String format,
        String dim,
        Center center,
        double radius,
        Summary summary,
        List<Entity> entities
) {
    public record Center(RMcpBlockPosData block, int chunkX, int chunkZ, int sectionY) {
    }

    public record EntityRef(String type, double distance) {
    }

    public record Summary(int total, int monsters, int animals, EntityRef nearestMonster, EntityRef nearestAnimal) {
    }

    public record Entity(
            String dim,
            String uuid,
            String type,
            String name,
            String category,
            RMcpPosData pos,
            double distance,
            Float health,
            Float maxHealth,
            boolean hostile,
            Boolean baby
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpNearbyEntitiesData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpNearbyEntitiesData(
    val dim: String,
    val center: Center,
    val radius: Double,
    val summary: Summary,
    val entities: List<Entity>
) {
    @JvmRecord
    data class Center(val block: RBlockPos, val chunkX: Int, val chunkZ: Int, val sectionY: Int)

    @JvmRecord
    data class EntityRef(val type: String, val distance: Double)

    @JvmRecord
    data class Summary(
        val total: Int,
        val monsters: Int,
        val animals: Int,
        val items: Int,
        val nearestMonster: EntityRef,
        val nearestAnimal: EntityRef,
        val nearestItem: EntityRef?
    )

    @JvmRecord
    data class Item(val snbt: String?)

    @JvmRecord
    data class Entity(
        val dim: String,
        val uuid: String,
        val type: String,
        val name: String,
        val category: String,
        val pos: REntityPosData,
        val distance: Double,
        val health: Float,
        val maxHealth: Float,
        val hostile: Boolean,
        val baby: Boolean,
        val item: Item?
    )
}
